lib/session.js

```diff
@@ -115,10 +133,25 @@ exports.signin = function(lookup, req, res, onSuccess, onFail) {
 
 exports.signout = function(req, res, next) {
 
-	res.clearCookie('keystone.uid');
-	req.user = null;
+	keystone.callHook(req.user, 'pre:signout', function(err) {
+		if (err) {
+			console.log("An error occurred in signout 'pre' middleware", err);
```

Comment:
@creynders sorry for the delay,  yes I think anything that more clearly outlines what is actually going on,  is always an improvement.